substrate/client/cli/src/commands/run_cmd.rs

```diff
@@ -410,15 +447,68 @@ impl CliConfiguration for RunCmd {
 			.into())
 	}
 
-	fn rpc_addr(&self, default_listen_port: u16) -> Result<Option<SocketAddr>> {
-		let interface = rpc_interface(
+	fn rpc_addr(&self, default_listen_port: u16) -> Result<Option<Vec<RpcEndpoint>>> {
+		if !self.rpc_endpoint.is_empty() {
+			for endpoint in &self.rpc_endpoint {
+				// Technically, `0.0.0.0` isn't a public IP address, but it's a way to listen on
+				// all interfaces. Thus, we consider it as a public endpoint and warn about it.
+				if endpoint.rpc_methods == RpcMethods::Unsafe && endpoint.is_global() ||
```

Comment:
//cc @BulatSaif @PierreBesson I haven't introduced any flag such as `--unsafe-external` and `--external` here, it's up to the user to define such a thing by the listen-addr but I added a similar warning log.

Do you want extra flags for that anyway? 

Now one need to do:  `--rpc-endpoint "listen-addr=0.0.0.0:9944,methods=unsafe"

but we could add something extra such as option for

` --rpc-endpoint "listen-addr=0.0.0.0:9944,methods=unsafe,unsafe-external=true"` but I don't see how that could make things better though...